### DISCUSSION TITLE
Installation of Velero on GCP

### DISCUSSION THREAD

**Discussion by sameershinde2409** [vmware-tanzu/velero#3248] [Community support Q&A]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)
velero install \
    --provider gcp \
    --plugins velero/velero-plugin-for-gcp:v1.0.1 \
    --bucket gs://sameer-test-velero-backup/ \
    --secret-file ./credentials-velero

**What did you expect to happen:**
An error occurred: fork/exec /plugins/velero-plugin-for-gcp: permission denied

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

An error occurred: fork/exec /plugins/velero-plugin-for-gcp: permission denied

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 1.4
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):1.18
- Kubernetes installer & version: 1.18
- Cloud provider or hardware configuration:GCP
- OS (e.g. from `/etc/os-release`):Ubuntu

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by zubron** [comment]
Hi @sameershinde2409! Sorry you're having issues with this. I'm not sure why there is an `permission denied` error when running the GCP plugin. Can you please provide the full logs for the velero deployment using `kubectl logs deployment/velero -n velero`? Do you have any pod security policies or other security settings in place? That may be affecting the permissions.

Another note, the version of the GCP plugin that you are using is not compatible with the version of Velero you are running. Please see the [GCP plugin compatibility table](https://github.com/vmware-tanzu/velero-plugin-for-gcp#compatibility). For Velero 1.4, you will need to use version 1.1.0 of the plugin. I don't know if this is the cause of your issue but you should ensure that the correct version is used.

**Comment by nrb** [comment]
I would recommend trying with `--bucket sameer-test-velero-backup`, removing the protocol information.

**Comment by sameershinde2409** [comment]
Thanks for your response. Here is the log output as requested. I did the changes as suggested.

velero install \ --provider gcp \ --plugins velero/velero-plugin-for-gcp:v1.1.0 \ --bucket gs://sameer-test-velero-backup/ \ --secret-file ./credentials-velero
Log-output:

time="2020-12-15T06:58:06Z" level=info msg="Starting Velero server v1.4.0 (5963650c9d64643daaf510ef93ac4a36b6483392)" logSource="pkg/cmd/server/server.go:179"time="2020-12-15T06:58:06Z" level=info msg="1 feature flags enabled []" logSource="pkg/cmd/server/server.go:181"time="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/crd-remap-versiontime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/podtime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pvtime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-accounttime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pv-from-pvctime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pvc-from-podtime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-pvc-node-selectortime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-storage-classtime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/cluster-role-bindingstime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/crd-preserve-fieldstime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/jobtime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/podtime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/restictime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/role-bindingstime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/servicetime="2020-12-15T06:58:06Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-accountAn error occurred: fork/exec /plugins/velero-plugin-for-gcp: permission denied

________________________________

**Comment by sameershinde2409** [comment]
Thanks for the reply, I have done the changes in bucket parameter as suggested but same error.

log output
time="2020-12-15T07:05:36Z" level=info msg="setting log-level to INFO" logSource="pkg/cmd/server/server.go:177"time="2020-12-15T07:05:36Z" level=info msg="Starting Velero server v1.4.0 (5963650c9d64643daaf510ef93ac4a36b6483392)" logSource="pkg/cmd/server/server.go:179"time="2020-12-15T07:05:36Z" level=info msg="1 feature flags enabled []" logSource="pkg/cmd/server/server.go:181"time="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/crd-remap-versiontime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/podtime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pvtime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-accounttime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pv-from-pvctime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pvc-from-podtime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-pvc-node-selectortime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-storage-classtime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/cluster-role-bindingstime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/crd-preserve-fieldstime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/jobtime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/podtime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/restictime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/role-bindingstime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/servicetime="2020-12-15T07:05:36Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-accountAn error occurred: fork/exec /plugins/velero-plugin-for-gcp: permission denied

________________________________

**Comment by zubron** [comment]
Thanks for the extra details, @sameershinde2409! Just asking again from an earlier comment: Do you have any pod security policies or other security settings in place? That may be affecting the permissions.

**Reply by ghost** [reply]
hello, I'm going through the same problem but using plugin for aws.

I'm not using pod security policies, I put the velero deployment in debug and this is the message I get debugged


[root@node1 velero]# kubectl logs deployment/velero -n velero
time="2021-06-17T18:51:39Z" level=info msg="setting log-level to DEBUG" logSource="pkg/cmd/server/server.go:172"
time="2021-06-17T18:51:39Z" level=info msg="Starting Velero server v1.6.0 (5bd70fd8eef316d220317245e46dc6016c348dce)" logSource="pkg/cmd/server/server.go:174"
time="2021-06-17T18:51:39Z" level=info msg="1 feature flags enabled []" logSource="pkg/cmd/server/server.go:176"
time="2021-06-17T18:51:39Z" level=debug msg="starting plugin" args="[/velero run-plugins --log-level debug --features ]" cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" path=/velero
time="2021-06-17T18:51:39Z" level=debug msg="plugin started" cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" path=/velero pid=13
time="2021-06-17T18:51:39Z" level=debug msg="waiting for RPC address" cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" path=/velero
time="2021-06-17T18:51:39Z" level=debug msg="Setting log level to DEBUG" cmd=/velero logSource="pkg/plugin/framework/server.go:199" pluginName=velero
time="2021-06-17T18:51:39Z" level=debug msg="plugin address" address=/tmp/plugin226824128 cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" network=unix pluginName=velero
time="2021-06-17T18:51:39Z" level=debug msg="using plugin" cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" version=2
time="2021-06-17T18:51:39Z" level=debug msg="plugin process exited" cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" path=/velero pid=13
time="2021-06-17T18:51:39Z" level=debug msg="plugin exited" cmd=/velero logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74"
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/crd-remap-version
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pv
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-account
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pv-from-pvc
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pvc-from-pod
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-pvc-node-selector
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-storage-class
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/cluster-role-bindings
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/crd-preserve-fields
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/init-restore-hook
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/job
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/restic
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/role-bindings
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service
time="2021-06-17T18:51:39Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-account
time="2021-06-17T18:51:39Z" level=debug msg="starting plugin" args="[/plugins/velero-plugin-for-aws --log-level debug --features ]" cmd=/plugins/velero-plugin-for-aws logSource="pkg/plugin/clientmgmt/logrus_adapter.go:74" path=/plugins/velero-plugin-for-aws
An error occurred: fork/exec /plugins/velero-plugin-for-aws: permission denied

**Comment by indrajithgihan** [comment]
@caiofabbio 
I am having the same problem with aws plugin. Did you manage to resolve this? If so how? Thanks

**Comment by indrajithgihan** [comment]
This is due to /var partition mounted as noexec (non-executable) in the worker node. I have remounted the /var partition with exec and it resolved the issue.

mount -o remount,exec /var

@sameershinde2409 
Please try above

**Reply by prasadjlv** [reply]
@indrajithgihan Thank you for the suggestion, do you know of any alternatives other than mounting /var with exec option? This suggestion is not an option because we apply CIS Benchmarks (https://www.cisecurity.org/benchmark/red_hat_linux) on the nodes and they all have /var mounted with noexec option.

**Reply by prasadjlv** [reply]
One option that has worked for us is to set emptyDir to Memory with a sizeLimit so that it does not use /var.
